packages/core/src/caches/index.ts

```diff
@@ -56,6 +45,112 @@ export class RedisCache implements CacheStore {
       consoleLog.info('[CACHE] Disconnected from Redis');
     }
   }
+
+  protected getSocketOptions(url: URL) {
+    const certFile = url.searchParams.get('cert');
+    const keyFile = url.searchParams.get('key');
+    const caFile = url.searchParams.get('certroot');
+
+    return {
+      rejectUnauthorized: yes(url.searchParams.get('reject_unauthorized')),
+      tls: url.protocol === 'rediss',
+      cert: certFile ? fs.readFileSync(certFile).toString() : undefined,
+      key: keyFile ? fs.readFileSync(keyFile).toString() : undefined,
+      ca: caFile ? fs.readFileSync(caFile).toString() : undefined,
+      reconnectStrategy: (retries: number, cause: Error) => {
+        if ('code' in cause && cause.code === 'SELF_SIGNED_CERT_IN_CHAIN') {
+          // This will throw only if reject unauthorized is true (default).
+          // Doesn't make sense to retry.
+          return false;
+        }
+
+        return Math.min(retries * 50, 1000);
+      },
+    };
+  }
+
+  protected abstract ping(): Promise<string | undefined>;
+}
+
+export class RedisCache extends RedisCacheBase {
+  readonly client?: RedisClientType;
+
+  constructor() {
```

Comment:
can we align this constructor with `RedisClusterCache`? i.e. use `constructor(connectionUrl: URL)`

Reply:
As the non-cluster client valid dsn can be only `"1"`, which is not a valid URL, the best I can do is to accept `connectionUrl: string | undefined`